src/rust-crypto/rust-crypto.ts

```diff
@@ -465,6 +470,25 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
         );
     }
 
+    /**
+     * Implementation of {@link CryptoApi#signObject}
+     */
+    public async signObject<T extends ISignableObject & object>(obj: T): Promise<void> {
```

Comment:
I need to update that it's not correct at the moment, but having this [rust PR available](https://github.com/matrix-org/matrix-rust-sdk/pull/2196) would make it easier to implement